src/game/collision-mediator.ts

```diff
@@ -0,0 +1,155 @@
+import { eventDestroyProjectile } from "../events/game-event";
+import { Vector } from "../utility/vector";
+import { GameContext } from "./game-context";
+import { Obstacle } from "./obstacle";
+import { Player } from "./player";
+import { Projectile } from "./projectile";
+
+export class CollisionMediator {
+    static processCollisions(context: GameContext) {
+        this.handleProjectileCollisionsWithPlayers(context);
+        this.handleProjectileCollisionsWithObstacles(context);
+        this.handlePlayerCollisionsWithObstacles(context);
+        this.handleObstacleCollisionsWithObstacles(context);
+    }
+
+    private static handleProjectileCollisionsWithPlayers(context: GameContext) {
+        context.projectiles.forEach((projectile) => {
+            context.players.forEach((player) => {
```

Comment:
Since we are looping through a lot of pairs of things all over the place, I propose adding a new utility funciton to DRY up the code a bit
```ts
function forEachPairOf<T1, T2>(
    arr1: T1[],
    arr2: T2[],
    fn: (e1: T1, e2: T2) => void
){
    arr1.forEach((e1) => {
        arr2.forEach((e2) => {
            fn(e1, e2)
        })
    })
}
```